core/pages/Category.js

```diff
@@ -148,8 +114,21 @@ export default {
       this.$bus.$off('user-after-logout', this.onUserPricesRefreshed)
     }
   },
-  beforeRouteUpdate (to, from, next) {
-    this.validateRoute(to)
+  async beforeRouteUpdate (to, from, next) {
+    if (from.path !== to.path) { // category changed - not just filters
+      try {
+        await this.$store.dispatch('category/fetchAsync', { route: to, context: null, slug: to.params.slug })
+      } catch (err) {
+        if (err.message.indexOf('query returned empty result') > 0) {
```

Comment:
```suggestion
        if (err.message.indexOf('query returned empty result') >= 0) {
```
https://rules.sonarsource.com/java/tag/suspicious/RSPEC-2692